drt_1d.py

```diff
@@ -1,147 +1,118 @@
 """
-Title: 1-D Geometry
-
-Description:
-This script is under construction. The goal is to approximate determinisitc radiation transport for a 1-D geometry.
-
-Contents:
-
-TODO:
-- deriving adjoint solution
-- adding obstacles
-- determining stopping criterion
-- create test problems
-
+Title: Deterministic Radiation Trasport - 1-D Toy Problem with Voxelized Geometry
 Author: Damian Jilk
-Date: 10/04/2023
 """
 
 import numpy as np
 import json
+import csv
 
-
-def calculate_distance(x_a: float, x_b: float):
+def construct_source_term(num_voxels:int, source_voxel_index:int, source_strength:float=1):
     """
-    Calculate the distance between two points along a 1-D geometry.
+    Constructs the source term vector. Assumes a point source. Minor modifications can be made to allow for a distribution.
 
-    Args:
-        x_a (float): The x-coordinate of point A.
-        x_b (float): The x-coordinate of point B.
+    Parameters:
+    - num_voxels (int): Number of spatial elements (voxels).
+    - source_voxel_index (int): Index of the voxel where the source is located.
+    - source_strength (float, optional): Strength of the source with default value of 1.
 
     Returns:
-        float: The absolute distance between point A and point B.
+    - numpy.ndarray: Source term vector.
     """
-    return abs(x_b - x_a)
-
-# Task 2.2: Construct optical distance along ray
-
-# Calculate macroscopic cross-section (Sigma_t_m)
+    s = np.zeros(num_voxels)
+    s[source_voxel_index] = source_strength
+    return s
 
-
-def calculate_macroscopic_cross_section(sigma_s_m: float, sigma_a_m: float):
+def calculate_K_trans(ri:int, rj:int, voxel_size:float, sigma_s_matrix:np.ndarray):
     """
-    Calculate the total cross section.
+    Calculate the neutron transition probability. Used by the construct_transition_matrix function to build the K_trans matrix.
```

Comment:
I think this may be best referred to as the streaming operator.  Based on what is returned, this is the probability that a particle born at the left side of `ri` streams to the left side of `rj` without interacting.